pkg/prometheus/promcfg.go

```diff
@@ -726,6 +726,38 @@ func (cg *ConfigGenerator) addProxyConfigtoYaml(
 	return cfg
 }
 
+func (cg *ConfigGenerator) addCustomHTTPConfigtoYaml(
+	cfg yaml.MapSlice,
+	store assets.StoreGetter,
+	customHTTPConfig monitoringv1.CustomHTTPConfig,
+) yaml.MapSlice {
+	if reflect.ValueOf(customHTTPConfig).IsZero() {
+		return cfg
+	}
+
+	if len(customHTTPConfig.HTTPHeaders) > 0 {
```

Comment:
I think we should validate versioning before iterate over HTTPHeaders just to avoid this effort

Reply:
Thanks @nicolastakashi 
I got it
Add prometheus version validate
ref: https://github.com/prometheus-operator/prometheus-operator/pull/7003/commits/da942738895416ed833c2646dac391ee7e5236be

Reply:
Hmmm!
But we are still validating only after the loop, maybe you don't push the fix?

Reply:
Thanks @nicolastakashi 
Got it. Add validate done.
ref: https://github.com/prometheus-operator/prometheus-operator/pull/7003/files#r1895200004